pkg/streamingpromql/query.go

```diff
@@ -82,23 +82,33 @@ func newQuery(ctx context.Context, queryable storage.Queryable, opts promql.Quer
 			return nil, fmt.Errorf("query expression produces a %s, but expression for range queries must produce an instant vector or scalar", parser.DocumentedType(expr.Type()))
 		}
 	}
+	q.root, err = q.convertToOperator(expr)
+	if err != nil {
+		return nil, err
+	}
 
+	return q, nil
+}
+
+func (q *Query) convertToOperator(expr parser.Expr) (types.Operator, error) {
+	var op types.Operator
+	var err error
 	switch expr.Type() {
 	case parser.ValueTypeMatrix:
-		q.root, err = q.convertToRangeVectorOperator(expr)
+		op, err = q.convertToRangeVectorOperator(expr)
```

Comment:
[nit] This (and the `case` below) could be simplified to this now this is outside `newQuery()`:

```suggestion
		return q.convertToRangeVectorOperator(expr)
```